orangecontrib/text/corpus_to_network.py

```diff
@@ -0,0 +1,292 @@
+        self.word_threshold = 0
+        self.freq_threshold = 0
+        self.word_network = None
+        self.document_network = None
+        self.word2ind = None
+        self.word_freqs = None
+        self.document_items = Table(corpus.domain,
+                                    corpus.X,
+                                    corpus.Y,
+                                    corpus.metas)
+        self.word_items = None
+        self.state = None
+        self.num_tokens = 0
+        self.last_called_nodes = True
+        self.param_cache = list()
+        self.mask = None
+
+    def __call__(self, document_nodes: bool = True,
+                 window_size: int = 1, threshold: int = 1,
+                 freq_threshold: int = 1,
+                 state: Optional[TaskState] = None) -> Network:
+        """Constructs network from a corpus. Network nodes can be either
+        documents or words. Two document nodes are connected if the number
+        of words (tokens) they share is larger (or equal to) than threshold.
+        Two words (tokens) co-occur if they are located inside a window
+        defined by window_size within a single document in corpus.
+        Two word nodes are connected if number of their co-occurences
+        is larger (or equal to) than threshold. In first case edge weight
+        represents number of words that two documents share and in second case
+        weight is number of co-occurences of two words (tokens)
+        in a corpus.
+
+        Parameters
+        ----------
+        document_nodes : bool
+            If True, the nodes of returned network will be documents, otherwise
+            words.
+        window_size: int
+            Size of window (actual size of window will be
+            2 * window_size + 1). Applies only if document_nodes=False.
+        threshold: int
+            Threshold that determines if edge between two nodes should exist.
+        freq_threshold: int
+            Ignore words with frequency smaller than freq_threshold.
+            Applies only if document_nodes=False.
+        state: TaskState
+            Optional parameter used in Orange widget. If using from script,
+            ignore this parameter.
+
+        Returns
+        -------
+        Network
+            Network obtained from corpus with given parameters.
+        Raises
+        ------
+        ValueError
+            If threshold <= 0 or window_size <= 0 or freq_threshold <= 0
+            or if document_nodes is not instance of bool.
+        """
+
+        if threshold <= 0:
+            raise ValueError("Threshold must be at least 1.")
+        if window_size <= 0:
+            raise ValueError("Window size must be at least 1.")
+        if freq_threshold <= 0:
+            raise ValueError("Frequency threshold must be at least 1.")
+        if not isinstance(document_nodes, bool):
+            raise ValueError("Document_nodes must be bool.")
+        
+        self.state = state
```

Comment:
It is not nice to store a parameter that appears on every call as a class attribute. I would suggest passing it to functions as an additional parameter. Since I see that state is used only for interesting and progress I suggest not passing the complete state to this function but preferably a callback function, such it is done in embedding - it is the more standard way. It is nicer that the callback function is created at the widget side (in runner).

Btw. Vesna learned me an even better way to interrupt the process. Before I was always making that with passion true/false out of the callback. An easier way is that callback directly raises an error and that way stops the processing.